src/payment/strategies/amazon-pay-v2/amazon-pay-v2-payment-strategy.ts

```diff
@@ -93,6 +93,16 @@ export default class AmazonPayV2PaymentStrategy implements PaymentStrategy {
 
         this._walletButton.click();
 
+        if (this._isModalFlow(region)) {
+            return new Promise((_, reject) => {
+                window.addEventListener(
```

Comment:
Given it's not explicit form the code itself why we're doing the focus event,
I'd suggest adding a comment for future developers. Something like:

```
// Focus of parent window used to try and detect the user cancelling the Amazon log in modal
// Should be refactored if/when Amazon add a modal close hook to their SDK 
```

Reply:
Agree.